rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/connection/DisconnectionRouter.java

```diff
@@ -49,27 +46,45 @@
          One of those events must happen eventually. Then the adapterStateObservable (which uses BroadcastReceiver on a Context) will
          get unsubscribed. The rest of this chain lives only in the @ConnectionScope context and will get Garbage Collected eventually.
          */
-        adapterMonitoringDisposable = awaitAdapterNotUsable(adapterWrapper, adapterStateObservable)
+        final Disposable adapterMonitoringDisposable = awaitAdapterNotUsable(adapterWrapper, adapterStateObservable)
                 .map(new Function<Boolean, BleException>() {
                     @Override
                     public BleException apply(Boolean isAdapterUsable) {
                         return BleDisconnectedException.adapterDisabled(macAddress);
                     }
                 })
-                .firstElement()
-                .subscribe(new Consumer<BleException>() {
+                .doOnNext(new Consumer<BleException>() {
                     @Override
-                    public void accept(BleException exception) {
+                    public void accept(BleException e) {
                         RxBleLog.d("An exception received, indicating that the adapter has became unusable.");
-                        exceptionOccurred = exception;
-                        notifySubscribersAboutException();
```

Comment:
Errors from `BluetoothAdapter` being switched off are now relayed through `bleExceptionBehaviorRelay`. The `bleExceptionBehaviorRelay` then passes the errors to `firstDisconnectionValueObs` and `firstDisconnectionExceptionObs` which are subscribed from the outside. Subscribers management is now done solely by RxJava

Reply:
Thank you for reply! 
I'm sorry. If you look closely, you are streaming to bleExceptionBehaviorRelay in subscribe...

Reply:
Yes, the `bleExceptionBehaviorRelay` should be subscribed to errors comming from turning off the BluetoothAdapter as soon as possible. It will also get unsubscribed at the first occurance of any error. The `bleExceptionBehaviorRelay` will keep the error for the future subscribers